### ISSUE TITLE
Speech Commands model throws exception with large number of Clients

### ISSUE THREAD

**Issue by rizhu** [rizhu/rayleaf-source#12]
Speech Commands should be able to support 2000 Clients and it runs fine for small numbers of Clients (e.g. 100). But when 2000 clients are used, an error arises with the max pooling.

To reproduce, make and run this experiment script in the root of the project. `gpus_per_client_cluster` and `num_client_clusters` may need to be changed depending on the setup. Generally, I use `CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 python3 script.py `. Final test accuracy should be at least 60%

```
from datetime import datetime

import rayleaf
from rayleaf.entities import Server, Client


rayleaf.run_experiment(
    dataset = "speech_commands",
    dataset_dir = "datasets/speech_commands/",
    output_dir=f"output/signavg-speech_commands-{curr_time}/",
    model = "m5",
    num_rounds = 100,
    eval_every = 10,
    ServerType=Server,
    client_types=[(Client, 2000)],
    clients_per_round = 40,
    client_lr = 0.06,
    batch_size = 64,
    seed = 0,
    use_val_set = False,
    num_epochs = 5,
    gpus_per_client_cluster=0.1,
    num_client_clusters=20,
    save_model=False
)
```
